src/main/java/fi/aalto/cs/intellij/services/CourseInformation.java

```diff
@@ -0,0 +1,106 @@
+package fi.aalto.cs.intellij.services;
+
+import com.intellij.openapi.components.ServiceManager;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CourseInformation {
+  private static final Logger logger = LoggerFactory
+      .getLogger(CourseInformation.class);
+
+  private String courseName;
+  private Map<String, URL> moduleUrls;
+
+  public CourseInformation() {
+    // Replace this with the correct path when testing with a local course configuration file.
+    parse("o1.json");
+  }
+
+  /**
+   * The CourseInformation instance managed by the service manager.
+   */
+  public static CourseInformation getInstance() {
+    return ServiceManager.getService(CourseInformation.class);
+  }
+
+  /**
+   * Parses the given course configuration file.
+   * @param pathToCourseConfig The path to the course configuration file.
+   */
+  public void parse(String pathToCourseConfig) {
+    logger.info("Parsing the course configuration file '{}'", pathToCourseConfig);
+    courseName = "";
+    moduleUrls = new HashMap<>();
+    FileReader file;
+    try {
+      file = new FileReader(pathToCourseConfig);
+    } catch (FileNotFoundException e) {
+      // TODO: notify the user of the error with a notification
```

Comment:
I propose to agree, at what level we start to notify users (students) about failing of the underlying processes. for example, if the read of the downloaded (read provided) file fails, can they really do something?

Reply:
Let's discuss this tomorrow :+1: